src/sarathi/llm/call_llm.py

```diff
@@ -37,8 +65,13 @@ def call_llm_model(prompt_info, user_msg, resp_type=None):
 
     Returns:
     """
+    url = retrieve_llm_url()
     url = "https://api.openai.com/v1/chat/completions"
-    model = prompt_info["model"]
+    try:
```

Comment:
@abhishek9sharma I think the logic would be simpler if we could instead use the `prompt_info["model"]` as the default when the environment variable is not defined. This can be accomplished by making `retrieve_model_name` take a parameter, which is the default like so:
```
def retrieve_model_name(default_model_name):
    """
    (some header comments here)
    """
    return get_env_var(["OPENAI_MODEL_NAME"], default=default_model_name)
```
and then replace this `try .. except ...` with just
```
model = retrieve_model_name(prompt_info["model"])
```

Reply:
Have modified in a way that a default models is always present, even if the user did not specify it in prompt_dict

Reply:
I relooked at this. As of now we are not allowing users to edit the prompt which we should allow at some point. https://github.com/abhishek9sharma/sarathi/issues/35

I think as of now best way would be get the model using this flow
- if present in env var use that
- elif not use the one in in prompt_dict
- else use a hardcoded value

Reply:
Thanks, @abhishek9sharma. Yes, this logic sounds good, but in the latest update, it seems to me the value of `prompt_info["model"]` will always be ignored because if `OPENAI_MODEL_NAME` is undefined, whatever the value of `prompt_info["model"]`, it will always be overridden with `gpt-4o-mini`.

To achieve the desired logic above, one way is to rewrite the following line in `retrieve_model_name`:
```
model_name = get_env_var(["OPENAI_MODEL_NAME"], default="gpt-4o-mini")
```
into
```
model_name = get_env_var(["OPENAI_MODEL_NAME"], default=(prompt_info["model"] if prompt_info["model"] else "gpt-4o-mini"))
```

Reply:
good catch fixed

Reply:
Thanks, @abhishek9sharma!